src/OpenTelemetry/Metrics/Implementation/Metric.cs

```diff
@@ -0,0 +1,59 @@
+﻿// <copyright file="Metric.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenTelemetry.Metrics.Implementation
+{
+    public class Metric<T>
```

Comment:
>T [](start = 24, length = 1)

just wondering, there is no any type constraints for the Metric?

Reply:
There will be.  This part is still work in progress and i will submit follow up prs.